prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -413,36 +408,72 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
-            } else {
-                val idTokenInfoString: String = client.post(restUrl) {
-                    expectSuccess = true // throw an exception if not successful
-                    contentType(ContentType.Application.Json)
+            val restUrl = restTransportInfo.authTokenUrl
+            val idTokenInfoString: String = client.post(restUrl) {
+                expectSuccess = true // throw an exception if not successful
+                postHeaders(
+                    if (restTransportInfo.authHeaders["Authorization-Type"] == "Basic Auth") {
+                        // Authorization-Type: "Basic Auth" requires the following:
+                        // Header:
+                        //  Authorization : "Basic + base64(username+password)"
+                        // Body:
+                        //  None
+                        // TODO: Add Authorization-Type: "Basic Auth" to DATAINGESTION (NBS) RESTTransport.authHeaders
+                        restTransportInfo.authHeaders.map { (key, value) ->
+                            if (key == "Authorization-Type" && value == "Basic Auth") {
+                                val credentialString = credential.user + ":" + credential.pass
+                                Pair(
+                                    "Authorization",
+                                    "Basic " + Base64.getEncoder().encodeToString(credentialString.encodeToByteArray())
+                                )
+                            } else {
+                                Pair(key, value)
+                            }
+                        }.toMap()
+                    } else {
+                        restTransportInfo.authHeaders
+                    }
+                )
+
+                // Authorization-Type: username/password requires the following:
+                // Header:
+                //  Content-Type: application/json
+                //  Authorization: username/password
+                // Body:
+                // { "username": "<username>", "password": "<password>"
+                    // LA-PHL.
+                    // TODO: Add Authorization-Type: "username/password" to LA-PHL RESTTransport.authHeaders
+                if (restTransportInfo.authHeaders["Authorization-Type"] == "username/password") {
+                    setBody(mapOf("username" to credential.user, "password" to credential.pass))
+                } else if (restTransportInfo.authHeaders["Authorization-Type"] == "email/password") {
+                        // Authorization-Type: email/password requires the following:
+                // Header:
+                //  Content-Type: application/json
+                //  Authorization: username/password
+                // Body:
+                // { "email": "<email@domain.com>", "password": "<password>"
+                    // OK-PHD.
+                    // TODO: Add Authorization-Type: "email/password" to OK-PHD RESTTransport.authHeaders
```

Comment:
This seems to be done for both staging and production settings.... is this referring to something else or can this TODO be removed?